x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/RBACEngine.java

```diff
@@ -680,78 +676,12 @@ public void getRemoteAccessRoleDescriptorsIntersection(
         }
 
         final Role role = ((RBACAuthorizationInfo) authorizationInfo).getRole();
-        final RemoteIndicesPermission remoteIndicesPermission;
-        try {
-            remoteIndicesPermission = role.remoteIndices().forCluster(remoteClusterAlias);
-        } catch (UnsupportedOperationException e) {
-            // TODO we will need to implement this to support API keys with assigned role descriptors
-            listener.onFailure(
-                new IllegalArgumentException(
-                    "cannot retrieve remote access role descriptors for API keys with assigned role descriptors.",
-                    e
-                )
-            );
-            return;
-        }
-
-        if (remoteIndicesPermission.remoteIndicesGroups().isEmpty()) {
+        final Collection<RoleDescriptor> remoteRoleDescriptors = role.getRemoteRoleDescriptors(remoteClusterAlias);
```

Comment:
I like the second suggestion. It made me realize I have a bug in the initial approach. Intersecting with an empty set would not result in an empty intersection. Returning a collection does not work in case one of the limited roles do not have remote indices.